nrf_security/src/drivers/nrf_oberon/Kconfig

```diff
@@ -135,7 +142,13 @@ config PSA_CRYPTO_DRIVER_ALG_GCM_OBERON
 	default y if !PSA_DEFAULT_OFF && PSA_WANT_ALG_GCM && \
 		     !PSA_CRYPTO_DRIVER_ALG_GCM_CC3XX
 
-# PSA_CRYPTO_DRIVER_ALG_HKDF_OBERON - Currently not supported
+config PSA_CRYPTO_DRIVER_ALG_HKDF_OBERON
+	bool
+	prompt "PSA HKDF support - oberon" if !PSA_PROMPTLESS
+	default y if !PSA_DEFAULT_OFF
```

Comment:
It is default y if HKDF is wanted. So I'm not sure there is anything to do here.

I removed the depends on PSA_OBERON_CORE here,  PSA_CRYPTO_DRIVER_HAS_KDF_SUPPORT_OBERON is the use of the PSA driver for key derivation, while HKDF is one of the supported key derivation algorithms.

Reply:
Discussed offline. 
CONFIG_PSA_CRYPTO_DRIVER_ALG_HKDF_CC3XX  is a non-function option that will be removed. But not in this PR.

CONFIG_PSA_CRYPTO_DRIVER_ALG_HKDF_OBERON will be enabled when Oberon drivers are enabled.
This is also always enabled in TF-M, because of TF-Ms dependency on a working key derivation driver.

Reply:
Created follow up task: NCSDK-21606